addon/-private/properties/alias.js

```diff
@@ -0,0 +1,95 @@
+import Ember from 'ember';
+
+import { getExecutionContext } from '../execution_context';
+import {
+  getProperty,
+  objectHasProperty
+} from '../helpers';
+
+/**
+ * Returns the value of some other property on the PageObject.
+ *
+ * @example
+ *
+ * const page = PageObject.create({
+ *   submitButton: {
+ *     scope: '.submit-button'
+ *   }
+ *   submit: alias('submitButton.click')
+ * });
+ *
+ * // calls `page.submitButton.click`
+ * page.submit();
+ *
+ * @example
+ *
+ * const page = PageObject.create({
+ *   submitButton: {
+ *     scope: '.submit-button'
+ *   }
+ *   isSubmitButtonVisible: alias('submitButton.isVisible')
+ * });
+ *
+ * // checks value of `page.submitButton.isVisible`
+ * assert.ok(page.isSubmitButtonVisible);
+ *
+ * @example
+ *
+ * const page = PageObject.create({
+ *   form: {
+ *     input {
+ *       scope: 'input'
+ *     },
+ *     submitButton: {
+ *       scope: '.submit-button'
+ *     }
+ *   },
+ *   fillFormInput: alias('form.input.fillIn', { chainable: true }),
+ *   submitForm: alias('form.submitButton.click', { chainable: true })
+ * });
+ *
+ * // executes `page.form.input.fillIn` then `page.form.submitButton.click`
+ * // and causes both methods to return `page` (instead of `page.form.input`
+ * // and `page.form.submitButton` respectively) so that the aliased methods
+ * // can be chained off `page`.
+ *
+ * page
+ *   .fillFormInput('foo')
+ *   .submitForm();
+ *
+ * @public
+ *
+ * @param {string} pathToProp - dot-separated path to a property specified on the PageObject
+ * @param {Object} options
+ * @param {Boolean} options.chainable - when this is true, an aliased
+ * method returns the PageObject node on which the alias is defined, rather
+ * than the PageObject node on which the aliased property is defined.
+ * @return {Descriptor}
+ *
+ * @throws Will throw an error if the PageObject does not have the specified property.
+ */
+export function alias(pathToProp, options = {}) {
+  return {
+    isDescriptor: true,
+
+    get(key) {
+      if (!objectHasProperty(this, pathToProp)) {
+        throw new Ember.Error(
```

Comment:
The easiest thing would be to merge #301 first so that I can build on those changes to `throwBetterErrors` here. Or, if this PR is ok to merge first, I can follow up with an improvement to the error messaging right after #301 merges.